test/statetest/statetest_loader.cpp

```diff
@@ -11,6 +11,12 @@ namespace evmone::test
 namespace json = nlohmann;
 using evmc::from_hex;
 
+template <>
+uint8_t from_json<uint8_t>(const json::json& j)
+{
+    return static_cast<uint8_t>(std::stoul(j.get<std::string>(), nullptr, 16));
```

Comment:
Since `stoul` returns `unsigned long` I suggest to do a runtime exception if it is larger than `uint8_t`.